test/integration/puppetlabs/services/certificate_authority/certificate_authority_int_test.clj

```diff
@@ -349,3 +350,54 @@
                          :else (do
                                  (Thread/sleep 500)
                                  (recur (dec times)))))))))))
+
+
+
+(deftest ^:integration revoke-compile-master-test
+  (testing "Revoking compile master certificate"
+    (logutils/with-test-logging
+      (bootstrap/with-puppetserver-running-with-mock-jrubies
+       "JRuby mocking is safe here because all of the requests are to the CA
+       endpoints, which are implemented in Clojure."
+        app
+        ;; Q: Does one need to make up other config params like for eg. infra-crl-path
+        ;; or can we assume anything unspecified will exist with appropriate defaults
+        ;; Would the below override whatever defaults?
+        {:certificate-authority {:compile-masters ["dummy-compile-master"]}}
```

Comment:
`bootstrap/with-puppetserver-running-with-mock-jrubies` will [use](https://jenkins-master-prod-1.delivery.puppetlabs.net/login?from=%2Fme%2Fconfigure) the configuration at `/dev/puppetserver.conf` with this used as an override for what's in there. Any additional overriding/defaults would be handled in the code you're writing.